test/modules/TestResultCoordinator/ITestResultComparer.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    /// <summary>
+    /// It is used to compare 2 test results whether they matches or not.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    interface ITestResultComparer<T>
+    {
+        bool Matches(T value1, T value2);
```

Comment:
Super nit: rename to Equals?

Reply:
I don't want to confuse with object.Equals method; this is just to tell that these 2 results matches.